src/Microsoft.Windows.CsWin32/Generator.cs

```diff
@@ -4351,14 +4364,89 @@ private ParameterSyntax CreateParameter(TypeHandleInfo parameterInfo, Parameter
                                 .WithLeadingTrivia(InlineArrayUnsafeAsSpanComment));
                 }
 
+                if (elementType is PredefinedTypeSyntax { Keyword: { RawKind: (int)SyntaxKind.CharKeyword } })
+                {
+                    // ...
+                    //     public string ToString(int length)
+                    //     {
+                    //         if (length < 0 || length > Length)
+                    //             throw new ArgumentOutOfRangeException(nameof(length), length, "Length must be between 0 and the fixed array length.");
+                    //         unsafe
```

Comment:
Why not ``AsSpan().Slice(0, length).ToString()`` instead of needing an unsafe block here and a fixed pointer?

Reply:
Because that requires `AsSpan` to exist. This was mentioned in the issue.

Reply:
not: Why not replace the `unsafe` block with an `unsafe` modifier on the method itself?

Reply:
> Saw something Jared said about it being a shame that C# conflates requiring the caller to be unsafe with enabling the method to do unsafe things, 

Using the `unsafe` modifier in the method signature has *no* impact on the caller, as the modifier is not represented in metadata. A caller's need to use `unsafe` is based on their own language rules based on the signature of the method. In C#, the caller only requires unsafe around a method call if the method signature involves pointers.

> I like keeping unsafe scoped as tightly as possible

Keeping in tightly scoped is fine, but in generated code where it's essentially the entire (short) method, it just means 4 more lines of code instead of a modifier. But it's generated code, and I don't too much care either.

Reply:
> Using the unsafe modifier in the method signature has no impact on the caller, as the modifier is not represented in metadata.

I understand how this works, and this is exactly what Jared was complaining about. It would have been cool if it did have an impact on the caller. This makes it slightly misleading when placed there.

Reply:
> It would have been cool if it did have an impact on the caller.

You think so? Why would you want to require the caller to use `unsafe` when they aren't dealing with pointers that would require it?

Reply:
A more generalized concept of `unsafe`. `IntPtr` APIs are much less safe than pointer APIs because it's the same consequences with less compiler checking.

Reply:
Threads https://twitter.com/jaredpar/status/795627645484437504 and https://twitter.com/jaredpar/status/1159164630696022022

> Enforcement is the big win here. Force consumers to acknowledge this method is dangerous w/o having to add a silly unused pointer param

Reply:
Anyway, that's a world that doesn't exist and I'm not here to prioritize my preferences, so let's go with `public unsafe string ToString` :+1: